Code/Tools/AssetProcessor/native/AssetManager/assetScannerWorker.cpp

```diff
@@ -118,14 +124,28 @@ void AssetScannerWorker::ScanForSourceFiles(const ScanFolderInfo& scanFolderInfo
         }
         else if (!AssetUtilities::IsInCacheFolder(absPath.toUtf8().constData())) // Ignore files in the cache
         {
-            //Entry is a file
-            m_fileList.insert(AZStd::move(assetFileInfo));
+            // Entry is a file - add it to the check list which will be processed multi-threaded later.
+            m_checkForExclusionList.insert(AZStd::move(assetFileInfo));
         }
     }
 }
 
 void AssetScannerWorker::EmitFiles()
 {
+    // multi-threaded processing of regular expresisons.  Note that each call to IsFileExcluded
+    // checks the file path against every exclusion rule, which turns out to be a decent chunk of work
```

Comment:
Nit: comment appears to be for the PR contributor, would suggest writing it for future maintainers. Not clear what "a decent chunk of work" means without context nor if this is a warning, general note etc. I would suggest removing or adding more context. Other comments of this nature have clear guidance to contributors.

Reply:
good idea, I'll reword or simplify